vuestorefront/models/product_template.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2023 ODOOGAP/PROMPTEQUATION LDA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+
+from odoo import api, fields, models
+from odoo.osv import expression
+from odoo.tools.sql import column_exists, create_column
+
+from odoo.addons.http_routing.models.ir_http import slug
+
+from ..utils import slugify_with_sfx
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _auto_init(self):
+        """Create/set slug via SQL to save on memory usage."""
+        cr = self.env.cr
+        if not column_exists(cr, "product_template", "slug"):
+            create_column(cr, "product_template", "slug", "varchar")
+            cr.execute("SELECT id, name FROM product_template WHERE name IS NOT NULL")
+            for res in cr.fetchall():
+                id_, name = res
+                cr.execute(
+                    "UPDATE product_template SET slug = %s WHERE id = %s",
+                    (slugify_with_sfx(name, f"-{id_}"), id_),
+                )
+        return super()._auto_init()
+
+    def _get_public_categ_slug(self, category_ids, category):
+        category_ids.append(category.id)
+
+        if category.parent_id:
+            category_ids = self._get_public_categ_slug(category_ids, category.parent_id)
+
+        return category_ids
```

Comment:
This method is confusing, as it seems it has nothing to do with slug. Maybe just rename it to `_get_category_with_parents` or sth.

Reply:
This is leftover from fork. Actually I no longer use it. We can remove this confusing method entirely :)